### PR TITLE
Add application health monitoring to VMSS resource processor 

### PR DESCRIPTION
**Pull request by anatbal**
# PR for issue #701 

- Add internal HTTP health endpoint to runner.py to verify it is running by adding a simple HTTP server inside the resource processor
-  Add health extension to VMSS on azure to have an health probe on port 8080

### CODE REVIEW

File: templates/core/terraform/resource_processor/vmss_porter/main.tf [hunk 1]
```diff
@@ -64,6 +64,23 @@ resource "azurerm_linux_virtual_machine_scale_set" "vm_linux" {
...
+    auto_upgrade_minor_version = false
+    automatic_upgrade_enabled  = false
+    name                       = "healthRepairExtension"
+    provision_after_extensions = []
+    publisher                  = "Microsoft.ManagedServices"
+    settings = jsonencode(
+      {
+        port        = 8080
+        protocol    = "http"
+        requestPath = "/get"
```
Comment (LizaShak) [inline templates/core/terraform/resource_processor/vmss_porter/main.tf]: /health

File: resource_processor/resources/httpserver.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
+from http.server import HTTPServer, BaseHTTPRequestHandler
+from socketserver import ThreadingMixIn
+
+
+class Handler(BaseHTTPRequestHandler):
```
Comment (LizaShak) [inline resource_processor/resources/httpserver.py:5]: RequestHandler

File: resource_processor/vmss_porter/runner.py [hunk 3]
```diff
@@ -200,7 +202,11 @@ async def runner():
...
 
-
 if __name__ == "__main__":
     logger_adapter.info("Started resource processor")
-    asyncio.run(runner())
+    asyncio.ensure_future(runner())
+    loop = asyncio.get_event_loop()
+    logger_adapter.info("Started http server")
+    httpserver_thread = threading.Thread(target=start_server)
+    httpserver_thread.start()
```
Comment (LizaShak) [inline resource_processor/vmss_porter/runner.py:211]: move line 209 to here

File: templates/core/terraform/resource_processor/vmss_porter/main.tf (hunk 1 above)
Comment (anatbal) [reply templates/core/terraform/resource_processor/vmss_porter/main.tf]: Fixed

File: resource_processor/resources/httpserver.py (hunk 2 above)
Comment (anatbal) [reply resource_processor/resources/httpserver.py]: Fixed

File: resource_processor/vmss_porter/runner.py [hunk 4]
```diff
@@ -200,7 +202,11 @@ async def runner():
             logger_adapter.info("All messages processed. Sleeping...")
             await asyncio.sleep(60)
 
-
 if __name__ == "__main__":
     logger_adapter.info("Started resource processor")
-    asyncio.run(runner())
+    asyncio.ensure_future(runner())
+    loop = asyncio.get_event_loop()
```
[the remark this answers is not in the document]
Comment (anatbal) [reply resource_processor/vmss_porter/runner.py]: Fixed

**Review by LizaShak — APPROVED** [review]

**Pull request merged**